internal/config/authentication.go

```diff
@@ -149,6 +149,13 @@ func (c *AuthenticationConfig) validate() error {
 	var sessionEnabled bool
 
 	for _, info := range c.Methods.AllMethods() {
+		if info.Method == auth.Method_METHOD_JWT {
```

Comment:
Rather than having to maintain as list here of methods which do or do not need their cleanup validating, we can lean into the info details on whether or not it requires a database:
```suggestion
		if !info.Enabled || !info.RequiresDatabase || info.Cleanup == nil {
```

We could also add this as a helper method onto [`AuthenticationMethodInfo`](https://github.com/flipt-io/flipt/blob/647eb5616b77d5d4151b4e0fb1eead90fae5e53f/internal/config/authentication.go#L312) (maybe `RequiresCleanup`) as this logic is also used in the `Config.ShouldRunCleanup` function above.